buildscripts/kokoro/cronet.sh

```diff
@@ -3,6 +3,119 @@
 set -exu -o pipefail
 cat /VERSION
 
-cd ./github/grpc-java/cronet
+BASE_DIR=$(pwd)
+
+# Set up APK size and dex count statuses
+
+gsutil cp gs://grpc-testing-secrets/github_credentials/oauth_token.txt ~/
+
+function set_status_to_fail_on_error {
+  cd $BASE_DIR/github/grpc-java
+
+  # TODO(ericgribkoff) Remove once merged
+  git checkout $KOKORO_GITHUB_PULL_REQUEST_COMMIT
+
+  ./buildscripts/set_github_status.py \
+    --sha1 $KOKORO_GITHUB_PULL_REQUEST_COMMIT \
+    --state error \
+    --description "Failed to calculate DEX count" \
+    --context android/dex_diff --oauth_file ~/oauth_token.txt
+  ./buildscripts/set_github_status.py \
+    --sha1 $KOKORO_GITHUB_PULL_REQUEST_COMMIT \
+    --state error \
+    --description "Failed to calculate APK size" \
+    --context android/apk_diff --oauth_file ~/oauth_token.txt
+}
+trap set_status_to_fail_on_error ERR
+
+$BASE_DIR/github/grpc-java/buildscripts/set_github_status.py \
+  --sha1 $KOKORO_GITHUB_PULL_REQUEST_COMMIT \
+  --state pending \
+  --description "Waiting for DEX count to be reported" \
+  --context android/dex_diff --oauth_file ~/oauth_token.txt
+$BASE_DIR/github/grpc-java/buildscripts/set_github_status.py \
+  --sha1 $KOKORO_GITHUB_PULL_REQUEST_COMMIT \
+  --state pending \
+  --description "Waiting for APK size to be reported" \
+  --context android/apk_diff --oauth_file ~/oauth_token.txt
+
+
+# Build Cronet
+
+cd $BASE_DIR/github/grpc-java/cronet
 ./cronet_deps.sh
 ../gradlew --include-build .. build
+
+
+# Install gRPC and codegen for the Android examples
+# (a composite gradle build can't find protoc-gen-grpc-java)
+
+cd $BASE_DIR/github/grpc-java
+
+export GRADLE_OPTS=-Xmx512m
+export PROTOBUF_VERSION=3.5.1
+export LDFLAGS=-L/tmp/protobuf/lib
+export CXXFLAGS=-I/tmp/protobuf/include
+export LD_LIBRARY_PATH=/tmp/protobuf/lib
+export OS_NAME=$(uname)
+
+# Proto deps
+buildscripts/make_dependencies.sh
+ln -s "/tmp/protobuf-${PROTOBUF_VERSION}/$(uname -s)-$(uname -p)" /tmp/protobuf
+
+./gradlew install
+
+cd ./examples/android/clientcache
+./gradlew build
+cd ../routeguide
+./gradlew build
+
+
+# Build and collect APK size and dex count stats for the helloworld example
+
+cd ../helloworld
+./gradlew build
+
+read -r ignored new_dex_count < \
+  <(${ANDROID_HOME}/tools/bin/apkanalyzer dex references app/build/outputs/apk/release/app-release-unsigned.apk)
+
+new_apk_size=$(stat --printf=%s app/build/outputs/apk/release/app-release-unsigned.apk)
+
+
+# Get the APK size and dex count stats using the target branch
+
+sudo apt-get install -y jq
+target_branch=$(curl -s https://api.github.com/repos/grpc/grpc-java/pulls/$KOKORO_GITHUB_PULL_REQUEST_NUMBER | jq -r .base.ref)
```

Comment:
Parsing JSON... that's unfortunate. I wonder if we could look at the merge commit parents. Also, it should probably be `.base.sha` since the ref may be different from when the clone was done.

Reply:
We want to compare against the base _this commit_ was built from. That is fixed at the start of the build. However, master can change over time, even during our build. If there is a merge commit, we could do `git rev-parse HEAD^` (or more unlikely `HEAD^2`, depending on how the merge was done). If there's no merge commit... maybe we could use `git merge-base`... But I sort of expect to see merge commits.

Reply:
Done. I was assuming that the base branch (e.g., master) was already in the local repo at the start of the build. That is not always the case, so if the checkout ends up pulling from the remote repository, using the base branch name could indeed end up pulling in changes that didn't exist when the build began.

It turns out that Github already creates the merge commit for you, and that can be used to build and test the PR: it's not quite documentation, but it appears to match reality: https://stackoverflow.com/questions/42626068/how-do-i-list-all-remote-refs

As far I can tell there will never be more than two parents of the merge commit, so we don't need to worry about `HEAD^2`.

cc @matt-kwong The `curl -s https://api.github.com/repos/grpc/grpc-java/pulls/$KOKORO_GITHUB_PULL_REQUEST_NUMBER | jq -r .base.ref` approach was copied from what grpc/grpc was doing. It looks like there might be a simpler and more correct alternative in `git rev-parse`  (Also, it is possible to change the target branch for a PR mid-build via Github's web UI, so the value being parsed from the JSON here could be different than what was used for the merge commit)